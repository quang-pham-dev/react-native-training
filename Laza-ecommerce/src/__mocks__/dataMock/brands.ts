/* A data array. */

export const brand = {
  id: '0',
  name: 'Nike',
  logoUrl:
    'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
};

export const brands = [
  {
    id: '0',
    name: 'Nike',
    logoUrl:
      'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
  },
  {
    id: '1',
    name: 'Adidas',
    logoUrl:
      'https://www.pikpng.com/pngl/m/42-429866_adidas-logo-png-images-free-download-rh-pngimg.png'
  },
  {
    id: '2',
    name: 'Fila',
    logoUrl:
      'https://toppng.com/uploads/preview/fila-logo-vector-download-free-11574216707jcmvfu58fy.png'
  },
  {
    id: '3',
    name: 'Puma',
    logoUrl:
      'https://banner2.cleanpng.com/20180613/gsj/kisspng-tracksuit-puma-logo-brand-clothing-puma-logo-5b20dc6b232426.487874081528880235144.jpg'
  },
  {
    id: '4',
    name: 'Gucci',
    logoUrl: 'https://esquilo.io/png/thumb/XBwczcMncYeff65-Vector-Gucci-Logo-PNG-Free-Download.png'
  }
];
