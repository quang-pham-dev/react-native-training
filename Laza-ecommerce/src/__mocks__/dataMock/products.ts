import { ProductProps } from 'types/Products';

export const product: ProductProps = {
  id: '0',
  brandId: '1',
  name: 'Nike',
  price: 100,
  type: 'shoes',
  like: true,
  title: 'Nike Sportswear Club Fleece',
  source:
    'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
  description:
    "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
  reviewer: {
    id: '2',
    name: 'John Doe',
    date: '13, Sep, 2020',
    image:
      'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
  },
  imageReview: [
    {
      id: '0',
      image:
        'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
    },
    {
      id: '1',
      image:
        'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
    },
    {
      id: '2',
      image:
        'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
    {
      id: '3',
      image:
        'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
    {
      id: '4',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    {
      id: '5',
      image:
        'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
  ],
  rating: '4.5',
  comment:
    'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
};

export const products: ProductProps[] = [
  {
    id: '0',
    brandId: '1',
    name: 'Nike',
    price: 100,
    type: 'shoes',
    like: true,
    title: 'Nike Sportswear Club Fleece',
    source:
      'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
    description:
      "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
    reviewer: {
      id: '2',
      name: 'John Doe',
      date: '13, Sep, 2020',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    imageReview: [
      {
        id: '0',
        image:
          'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
      },
      {
        id: '1',
        image:
          'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
      },
      {
        id: '2',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '3',
        image:
          'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '4',
        image:
          'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
      },
      {
        id: '5',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
    ],
    rating: '4.5',
    comment:
      'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
  },
  {
    id: '0',
    brandId: '1',
    name: 'Nike',
    price: 100,
    type: 'shoes',
    like: true,
    title: 'Nike Sportswear Club Fleece',
    source:
      'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
    description:
      "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
    reviewer: {
      id: '2',
      name: 'John Doe',
      date: '13, Sep, 2020',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    imageReview: [
      {
        id: '0',
        image:
          'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
      },
      {
        id: '1',
        image:
          'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
      },
      {
        id: '2',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '3',
        image:
          'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '4',
        image:
          'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
      },
      {
        id: '5',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
    ],
    rating: '4.5',
    comment:
      'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
  },
];
