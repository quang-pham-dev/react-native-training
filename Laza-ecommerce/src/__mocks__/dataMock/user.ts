export const user = {
  id: '621c4188dece66a40dad2951',
  avatar: 'https://picsum.photos/200',
  username: 'Quangpham',
  gender: 'male',
  email: 'admin@admin.com',
  phone: '+84 (919) 507231',
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  role: 'admin',
  password: '123456'
};
