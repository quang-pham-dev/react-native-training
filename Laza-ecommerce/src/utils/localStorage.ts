import AsyncStorage from '@react-native-async-storage/async-storage';

export const get = async (key: string) => await AsyncStorage.getItem(key);

export const set = async (key: string, data: string) => await AsyncStorage.setItem(key, data);

export const remove = async (key: string) => await AsyncStorage.removeItem(key);

export const setJSON = (key: string, data: any) => set(key, JSON.stringify(data));

export const getJSON = <T = any>(key: string): T | null => {
  const data = get(key);
  return typeof data === 'string' ? JSON.parse(data) : null;
};

export default { get, set, remove, setJSON, getJSON };
