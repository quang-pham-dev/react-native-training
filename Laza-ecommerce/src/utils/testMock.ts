export const navigationMock: any = {
  addListener: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  dismiss: jest.fn(),
  dispatch: jest.fn(),
  emit: jest.fn(),
  getParam: jest.fn(),
  goBack: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  isFocused: jest.fn(),
  navigate: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  router: undefined,
  setParams: jest.fn(),
  route: undefined,
  state: {
    key: '',
    routeName: '',
    params: {},
    path: '',
    index: 1,
    isTransitioning: false,
    routes: []
  }
};
