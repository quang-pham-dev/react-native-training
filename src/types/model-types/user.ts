export interface IUser {
  id: string
  avatar: string
  username: string
  password: string
  gender: string
  email: string
  phone: string
  address: string
  role: string
}

export interface UserResponse {
  access_token: string
  user: IUser
}

export interface LoginCredentials {
  username: string
  password: string
}
