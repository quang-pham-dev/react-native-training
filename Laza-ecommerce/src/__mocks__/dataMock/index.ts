import { user } from './user';
import { brand, brands } from './brands';
import { product, products } from './products';

export { brand, brands, product, products, user };
