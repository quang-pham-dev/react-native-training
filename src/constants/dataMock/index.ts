import {user} from './user'
import {brand, brands} from './brands'

export {brand, brands, user}
