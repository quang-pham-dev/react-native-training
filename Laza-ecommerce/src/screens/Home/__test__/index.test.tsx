import React from 'react';

// LIBS
import { cleanup, render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

// Screen
import HomeScreen from 'screens/Home';

// Utils
import { navigationMock } from 'utils/testMock';

// Mocks
import { user } from '__mocks__/dataMock/user';

describe('Home Screen', () => {
  const helloText = 'Hello';
  const wellComeText = 'Welcome to Laza.';

  let tree: any;
  const props = {
    navigation: navigationMock,
    route: {
      params: {}
    },
    currentUser: user,
    username: user.username
  };
  beforeEach(() => {
    tree = renderer.create(<HomeScreen {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  test('Should renders correctly', () => {
    const component = tree.toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Should render a Hello text', () => {
    const tree = render(<HomeScreen {...props} />);
    const { getByText } = tree;

    expect(getByText(helloText)).toBeTruthy();
  });

  test('Should render a Welcome to Laza. text', () => {
    const tree = render(<HomeScreen {...props} />);
    const { getByText } = tree;

    expect(getByText(wellComeText)).toBeTruthy();
  });
});
