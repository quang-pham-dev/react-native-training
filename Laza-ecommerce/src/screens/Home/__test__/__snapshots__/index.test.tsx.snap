// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Should renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 0,
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 45,
              "opacity": 1,
              "width": 45,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={Object {}}
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 45,
              "opacity": 1,
              "width": 45,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={Object {}}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "opacity": 1,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Inter_500Medium",
              "fontSize": 17,
              "lineHeight": 16,
              "textTransform": "none",
            },
            Array [
              Object {
                "fontFamily": "Inter_600SemiBold",
                "fontSize": 28,
                "lineHeight": 30,
              },
              undefined,
            ],
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        Hello
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
          "opacity": 1,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Inter_500Medium",
              "fontSize": 17,
              "lineHeight": 16,
              "textTransform": "none",
            },
            Array [
              Object {
                "color": "#8F959E",
                "fontFamily": "Inter_400Regular",
                "fontSize": 15,
                "lineHeight": 16,
              },
              undefined,
            ],
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        Welcome to Laza.
      </Text>
    </View>
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "translateY": 0,
            },
            Object {
              "scaleX": 1,
            },
            Object {
              "scaleY": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F5F6FA",
                "borderRadius": 10,
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "opacity": 1,
                  "width": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              />
            </View>
            <TextInput
              allowFontScaling={true}
              autoCorrect={false}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Search.."
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "paddingLeft": 12,
                    "paddingRight": 50,
                    "width": "100%",
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginLeft": 10,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "alignSelf": "center",
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 200,
      }
    }
  >
    <View
      style={
        Object {
          "opacity": 1,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingBottom": 15,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Inter_500Medium",
                "fontSize": 17,
                "lineHeight": 16,
                "textTransform": "none",
              },
              Array [
                Object {
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 17,
                  "lineHeight": 16,
                },
                undefined,
              ],
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          Choose Brand
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Inter_500Medium",
                "fontSize": 17,
                "lineHeight": 16,
                "textTransform": "none",
              },
              Array [
                Object {
                  "color": "#8F959E",
                  "fontFamily": "Inter_400Regular",
                  "fontSize": 13,
                  "lineHeight": 14,
                },
                undefined,
              ],
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          View All
        </Text>
      </View>
      <RNCSafeAreaView>
        <ActivityIndicator
          animating={true}
          color="#FFC107"
          hidesWhenStopped={true}
          size="small"
          style={
            Object {
              "margin": 13,
            }
          }
        />
      </RNCSafeAreaView>
      <View
        style={Object {}}
      >
        <RCTScrollView
          ListEmptyComponent={[Function]}
          ListFooterComponent={[Function]}
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReached={null}
          onEndReachedThreshold={0.5}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "Inter_500Medium",
                    "fontSize": 17,
                    "lineHeight": 16,
                    "textTransform": "none",
                  },
                  undefined,
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              No Brand found
            </Text>
            <View
              onLayout={[Function]}
            />
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginTop": 15,
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Inter_500Medium",
                "fontSize": 17,
                "lineHeight": 16,
                "textTransform": "none",
              },
              Array [
                Object {
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 17,
                  "lineHeight": 16,
                },
                undefined,
              ],
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          New Arrival
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Inter_500Medium",
                "fontSize": 17,
                "lineHeight": 16,
                "textTransform": "none",
              },
              Array [
                Object {
                  "color": "#8F959E",
                  "fontFamily": "Inter_400Regular",
                  "fontSize": 13,
                  "lineHeight": 14,
                },
                undefined,
              ],
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          View All
        </Text>
      </View>
      <RNCSafeAreaView>
        <ActivityIndicator
          animating={true}
          color="#FFC107"
          hidesWhenStopped={true}
          size="small"
          style={
            Object {
              "margin": 13,
            }
          }
        />
      </RNCSafeAreaView>
      <View
        style={
          Object {
            "marginTop": 15,
          }
        }
      >
        <RCTScrollView
          ListEmptyComponent={[Function]}
          ListFooterComponent={[Function]}
          ListFooterComponentStyle={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginBottom": 15,
            }
          }
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={6}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReached={null}
          onEndReachedThreshold={0.5}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={16}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "Inter_500Medium",
                    "fontSize": 17,
                    "lineHeight": 16,
                    "textTransform": "none",
                  },
                  undefined,
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              No products found
            </Text>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginBottom": 15,
                }
              }
            />
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
